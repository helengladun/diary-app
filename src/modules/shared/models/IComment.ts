export interface IComment {
    id: number,
    description: string,
    post_id: number
}