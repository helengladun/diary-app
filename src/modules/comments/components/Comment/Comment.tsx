import React, {Component} from 'react';
import {IComment} from "../../../shared/models/IComment";

interface IProps {
    data: IComment
}

export class Comment extends Component<IProps> {
    render() {
        const {data} = this.props;
        return (
            <div className="comment">
                <div className="square comment__colored-square"></div>
                <div className="comment__description">{data.description}</div>
            </div>
        );
    }
}