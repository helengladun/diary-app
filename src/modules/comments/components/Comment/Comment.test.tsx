import React from 'react';
import {shallow} from 'enzyme';
import {Comment} from "./Comment";
import {IComment} from "../../../shared/models/IComment";

describe('Comment component', () => {
    const data: IComment = {
        id: 1,
        description: 'some text',
        post_id: 2
    };


    test('should render Comment component correctly', () => {
        const wrapper = shallow(<Comment data={data}/>);
        expect(wrapper).toMatchSnapshot();
    });
});