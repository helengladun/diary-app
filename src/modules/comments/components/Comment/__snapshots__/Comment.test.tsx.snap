// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component should render Comment component correctly 1`] = `
<div
  className="comment"
>
  <div
    className="square comment__colored-square"
  />
  <div
    className="comment__description"
  >
    some text
  </div>
</div>
`;
